smartsim/_core/entrypoints/telemetrymonitor.py

```diff
@@ -0,0 +1,677 @@
+    entity.step_id = str(metadata.get("step_id") or "")
+    entity.task_id = str(metadata.get("task_id") or "")
+    entity.timestamp = int(persistable_entity.get("timestamp", "0"))
+    entity.path = str(exp_dir)
+    entity.status_dir = str(status_dir)
+
+    return entity
+
+
+def hydrate_persistable(
+    entity_type: str,
+    persistable_entity: t.Dict[str, t.Any],
+    exp_dir: pathlib.Path,
+) -> t.List[JobEntity]:
+    """Map entity data persisted in a manifest file to an object"""
+    entities = []
+
+    # an entity w/parent key creates persistables for entities it contains
+    parent_keys = {"shards", "models"}
+    parent_keys = parent_keys.intersection(persistable_entity.keys())
+    if parent_keys:
+        container = "shards" if "shards" in parent_keys else "models"
+        child_type = "orchestrator" if container == "shards" else "model"
+        for child_entity in persistable_entity[container]:
+            entity = _hydrate_persistable(child_entity, child_type, str(exp_dir))
+            entities.append(entity)
+
+        return entities
+
+    entity = _hydrate_persistable(persistable_entity, entity_type, str(exp_dir))
+    entities.append(entity)
+    return entities
+
+
+def hydrate_persistables(
+    entity_type: str,
+    run: t.Dict[str, t.Any],
+    exp_dir: pathlib.Path,
+) -> t.Dict[str, t.List[JobEntity]]:
+    """Map a collection of entity data persisted in a manifest file to an object"""
+    persisted: t.Dict[str, t.List[JobEntity]] = {
+        "model": [],
+        "orchestrator": [],
+    }
+    for item in run[entity_type]:
+        entities = hydrate_persistable(entity_type, item, exp_dir)
+        for new_entity in entities:
+            persisted[new_entity.type].append(new_entity)
+
+    return persisted
+
+
+def hydrate_runs(
+    persisted_runs: t.List[t.Dict[str, t.Any]], exp_dir: pathlib.Path
+) -> t.List[Run]:
+    """Map run data persisted in a manifest file to an object"""
+    the_runs: t.List[Run] = []
+    for run_instance in persisted_runs:
+        run_entities: t.Dict[str, t.List[JobEntity]] = {
+            "model": [],
+            "orchestrator": [],
+            "ensemble": [],
+        }
+
+        for key in run_entities:
+            _entities = hydrate_persistables(key, run_instance, exp_dir)
+            for entity_type, new_entities in _entities.items():
+                if new_entities:
+                    run_entities[entity_type].extend(new_entities)
+
+        run = Run(
+            run_instance["timestamp"],
+            run_entities["model"],
+            run_entities["orchestrator"],
+            run_entities["ensemble"],
+        )
+        the_runs.append(run)
+
+    return the_runs
+
+
+def load_manifest(file_path: str) -> t.Optional[RuntimeManifest]:
+    """Load a persisted manifest and return the content"""
+    manifest_dict: t.Optional[t.Dict[str, t.Any]] = None
+    try_count = 1
+
+    while manifest_dict is None and try_count < 6:
```

Comment:
Let's make `try_count` a module variable at least

Reply:
`try_count` or move the `6` to a module level `MAX_TRY_COUNT` variable? I don't really want to move `try_count` outside the scope of this fn if this fn is going to mutate the variable.